File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/shared/licensevideo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Cookie } from "./cookie";


export enum LicenseVideoSizeEnum {
    Web = "web",
    Sd = "sd",
    Hd = "hd",
    Fourk = "4k"
}


// LicenseVideo
/** 
 * Data required to license a video
**/
export class LicenseVideo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=auth_cookie" })
  authCookie?: Cookie;

  @SpeakeasyMetadata({ data: "json, name=editorial_acknowledgement" })
  editorialAcknowledgement?: boolean;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=price" })
  price?: number;

  @SpeakeasyMetadata({ data: "json, name=search_id" })
  searchId?: string;

  @SpeakeasyMetadata({ data: "json, name=show_modal" })
  showModal?: boolean;

  @SpeakeasyMetadata({ data: "json, name=size" })
  size?: LicenseVideoSizeEnum;

  @SpeakeasyMetadata({ data: "json, name=subscription_id" })
  subscriptionId?: string;

  @SpeakeasyMetadata({ data: "json, name=video_id" })
  videoId: string;
}

File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/shared/licensevideo.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import cookie as shared_cookie

class LicenseVideoSizeEnum(str, Enum):
    WEB = "web"
    SD = "sd"
    HD = "hd"
    FOURK = "4k"


@dataclass_json
@dataclasses.dataclass
class LicenseVideo:
    r"""LicenseVideo
    Data required to license a video
    """
    
    video_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('video_id') }})
    auth_cookie: Optional[shared_cookie.Cookie] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('auth_cookie') }})
    editorial_acknowledgement: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('editorial_acknowledgement') }})
    metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    price: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('price') }})
    search_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('search_id') }})
    show_modal: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('show_modal') }})
    size: Optional[LicenseVideoSizeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('size') }})
    subscription_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subscription_id') }})
    
